Add Timolia

l3nnartt opened the pull request:
Add Timolia.de

l3nnartt committed:
Add Timolia

code-factor committed:
[CodeFactor] Apply fixes to commit 73de44e

l3nnartt committed:
Remove unused Lines

Bas950 reviewed (changes requested):
Fix all errors, upload proof of it working

l3nnartt committed:
Merge branch 'PreMiD:master' into master

l3nnartt committed:
Add Imgur Link

l3nnartt committed:
== to ===

l3nnartt commented:
![Screenshot 2021-08-07 000937](https://user-images.githubusercontent.com/69513317/128575709-1901212e-5f9a-48e9-9377-3c16bad1764f.png)
![Screenshot 2021-08-07 001014](https://user-images.githubusercontent.com/69513317/128575749-550b95af-5563-45d5-89a8-eb8f51860d1f.png)
![Screenshot 2021-08-07 001057](https://user-images.githubusercontent.com/69513317/128575797-af3ac256-a818-4811-a2fe-35e021b24cfc.png)
![Screenshot 2021-08-07 001124](https://user-images.githubusercontent.com/69513317/128575845-19c9500e-1374-42c7-9b93-6a626d5ecbef.png)

l3nnartt committed:
test

l3nnartt committed:
Clean Code

code-factor committed:
[CodeFactor] Apply fixes to commit 94bbcb6

Bas950 commented on websites/T/timolia/dist/metadata.json:
```diff
@@ -0,0 +1,25 @@
+    "service": "Timolia",
+    "author": {
+      "name": "Lennart™",
+      "id": "0001"
```
Id should be your Discord ID, not your discriminator. 
Example (mine): 241278257335500811

Bas950 commented on websites/T/timolia/presence.ts:
```diff
@@ -0,0 +1,73 @@
+const presence = new Presence({
+    clientId: "872421983554502717"
+  }),
+
```
```suggestion
```

Bas950 commented on websites/T/timolia/presence.ts:
```diff
@@ -0,0 +1,73 @@
+presence.on("UpdateData", () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "large_image",
+    details: document.location.href,
```
details to href????

l3nnartt replied:
href displays the current website url

Slowlife01 replied:
*But why use it on details*

l3nnartt replied:
If you are on a website where there is no explicit description only the URL is shown

Bas950 replied:
Then something like "Unsupported Page" or just nothing is better in my opinion :/

l3nnartt replied:
alright, then i remove it :)

Bas950 commented on websites/T/timolia/presence.ts:
```diff
@@ -0,0 +1,73 @@
+
+  if (document.location.hostname === "www.timolia.de") {
+    if (document.location.pathname === "/") {
+        presenceData.startTimestamp = browsingStamp;
```
```suggestion
```
unneed, set at line 11

Bas950 commented on websites/T/timolia/presence.ts:
```diff
@@ -0,0 +1,73 @@
+        presenceData.smallImageKey = "minecraft";
+        presenceData.smallImageText = `${player} online players`;
+    } else if (document.location.pathname.startsWith("/stats")) {
+        presenceData.startTimestamp = browsingStamp;
```
```suggestion
```

Bas950 commented on websites/T/timolia/presence.ts:
```diff
@@ -0,0 +1,73 @@
+        presenceData.details = "Viewing stats from:";
+        presenceData.state = stats;
+    } else if (document.location.pathname.includes("/games")) {
+        presenceData.startTimestamp = browsingStamp;
```
```suggestion
```

Bas950 commented on websites/T/timolia/presence.ts:
```diff
@@ -0,0 +1,73 @@
+        presenceData.startTimestamp = browsingStamp;
+        presenceData.details = "Viewing gamemodes";
+    } else if (document.location.pathname.includes("/account")) {
+        presenceData.startTimestamp = browsingStamp;
```
```suggestion
```

Bas950 commented on websites/T/timolia/presence.ts:
```diff
@@ -0,0 +1,73 @@
+        presenceData.details = "Viewing account settings";
+    }
+  } else if (document.location.hostname === "howto.timolia.de") {
+        presenceData.startTimestamp = browsingStamp;
```
```suggestion
```

Bas950 commented on websites/T/timolia/presence.ts:
```diff
@@ -0,0 +1,73 @@
+        presenceData.details = "HowTo - Reading:";
+        presenceData.state = document.querySelector("body > div.md-container > main > div > div.md-content > article > h1").textContent;
+    } else if (document.location.hostname === "shop.timolia.de") {
+        presenceData.startTimestamp = browsingStamp;
```
```suggestion
```

Bas950 commented on websites/T/timolia/presence.ts:
```diff
@@ -0,0 +1,73 @@
+        presenceData.startTimestamp = browsingStamp;
+        presenceData.details = "Shop - Viewing store page";
+        if (document.location.pathname.startsWith("/checkout")) {
+            presenceData.startTimestamp = browsingStamp;
```
```suggestion
```

Bas950 commented on websites/T/timolia/presence.ts:
```diff
@@ -0,0 +1,73 @@
+        if (document.location.pathname.startsWith("/checkout")) {
+            presenceData.startTimestamp = browsingStamp;
+            presenceData.details = "Shop - Viewing shopping cart:";
+            presenceData.state = document.querySelector("body > div.sections-wrapper > section > div > div > div.jumbotron > div > div.packages > form > table > tbody > tr > td.col-md-6").textContent;
```
What exactly would it show here?

l3nnartt replied:
The article you are looking at

l3nnartt replied:
![Screenshot 2021-08-10 000212](https://user-images.githubusercontent.com/69513317/128780150-c3788b92-491d-4528-a1d5-a2100e50d7af.png)

Bas950 commented on websites/T/timolia/presence.ts:
```diff
@@ -0,0 +1,73 @@
+        presenceData.smallImageKey = "gitlab";
+    }
+
+    if (presenceData.details === null) {
```
```suggestion
    if (!presenceData.details) {
```

Bas950 commented on websites/T/timolia/presence.ts:
```diff
@@ -0,0 +1,73 @@
+    if (presenceData.details === null) {
+        presence.setTrayTitle();
+        presence.setActivity();
+    } else 
+        presence.setActivity(presenceData);
+    
```
```suggestion
    } else presence.setActivity(presenceData);
```

l3nnartt committed:
Merge branch 'PreMiD:master' into master

l3nnartt committed:
Update websites/T/timolia/presence.ts

Co-authored-by: Bas950 <bas.van.zanten007@gmail.com>

l3nnartt committed:
Update websites/T/timolia/presence.ts

Co-authored-by: Bas950 <bas.van.zanten007@gmail.com>

l3nnartt committed:
Update websites/T/timolia/presence.ts

Co-authored-by: Bas950 <bas.van.zanten007@gmail.com>

l3nnartt committed:
Update websites/T/timolia/presence.ts

Co-authored-by: Bas950 <bas.van.zanten007@gmail.com>

l3nnartt committed:
Update websites/T/timolia/presence.ts

Co-authored-by: Bas950 <bas.van.zanten007@gmail.com>

l3nnartt committed:
Update websites/T/timolia/presence.ts

Co-authored-by: Bas950 <bas.van.zanten007@gmail.com>

l3nnartt committed:
Update websites/T/timolia/presence.ts

Co-authored-by: Bas950 <bas.van.zanten007@gmail.com>

l3nnartt committed:
Update metadata.json

l3nnartt committed:
Update websites/T/timolia/presence.ts

Co-authored-by: Bas950 <bas.van.zanten007@gmail.com>

l3nnartt committed:
Update websites/T/timolia/presence.ts

Co-authored-by: Bas950 <bas.van.zanten007@gmail.com>

l3nnartt committed:
Update websites/T/timolia/presence.ts

Co-authored-by: Bas950 <bas.van.zanten007@gmail.com>

code-factor committed:
[CodeFactor] Apply fixes to commit 0d35731

l3nnartt committed:
Merge branch 'PreMiD:master' into master

dependabot[bot] committed:
🔼 Bump fastify/github-action-merge-dependabot from 2.1.1 to 2.4.0

Bumps [fastify/github-action-merge-dependabot](https://github.com/fastify/github-action-merge-dependabot) from 2.1.1 to 2.4.0.
- [Release notes](https://github.com/fastify/github-action-merge-dependabot/releases)
- [Commits](https://github.com/fastify/github-action-merge-dependabot/compare/v2.1.1...v2.4.0)

---
updated-dependencies:
- dependency-name: fastify/github-action-merge-dependabot
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
🔼 Bump mongodb from 3.6.9 to 4.1.0

Bumps [mongodb](https://github.com/mongodb/node-mongodb-native) from 3.6.9 to 4.1.0.
- [Release notes](https://github.com/mongodb/node-mongodb-native/releases)
- [Changelog](https://github.com/mongodb/node-mongodb-native/blob/4.1/docs/CHANGES_4.0.0.md)
- [Commits](https://github.com/mongodb/node-mongodb-native/compare/v3.6.9...v4.1.0)

---
updated-dependencies:
- dependency-name: mongodb
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

l3nnartt committed:
Merge branch 'PreMiD:master' into master

l3nnartt committed:
Merge pull request #1 from l3nnartt/dependabot/github_actions/fastify/github-action-merge-dependabot-2.4.0

🔼 Bump fastify/github-action-merge-dependabot from 2.1.1 to 2.4.0

l3nnartt committed:
Remove URL on Details

l3nnartt committed:
Update dependabot.yml

l3nnartt committed:
Update dependabot.yml

l3nnartt committed:
Merge pull request #10 from l3nnartt/dependabot/npm_and_yarn/mongodb-4.1.0

🔼 Bump mongodb from 3.6.9 to 4.1.0

Bas950 reviewed (changes requested):
I guess update your fork or make sure these changes are gone 
![image](https://user-images.githubusercontent.com/34792838/128909966-1071ab58-0c0d-4f4e-9529-7d8e28e428fc.png)

l3nnartt committed:
Merge branch 'PreMiD:master' into master

l3nnartt committed:
Revert "🔼 Bump mongodb from 3.6.9 to 4.1.0"

l3nnartt committed:
Merge pull request #11 from l3nnartt/revert-10-dependabot/npm_and_yarn/mongodb-4.1.0

Revert "🔼 Bump mongodb from 3.6.9 to 4.1.0"

l3nnartt committed:
Revert "🔼 Bump fastify/github-action-merge-dependabot from 2.1.1 to 2.4.0"

l3nnartt committed:
Merge pull request #12 from l3nnartt/revert-1-dependabot/github_actions/fastify/github-action-merge-dependabot-2.4.0

Revert "🔼 Bump fastify/github-action-merge-dependabot from 2.1.1 to 2.4.0"

l3nnartt commented:
oh sorry, i reverted the changes!

Bas950 commented on websites/T/timolia/dist/metadata.json:
```diff
@@ -0,0 +1,25 @@
+{
+    "$schema": "https://schemas.premid.app/metadata/1.3",
+    "service": "Timolia",
```
Service name and folder name must be the same (caps included)

l3nnartt replied:
fixed

l3nnartt committed:
Service Name

l3nnartt reviewed (commented):
fixed service name

Bas950 reviewed (commented):
Shouldn't the T be in uppercase?

l3nnartt committed:
Uppercase Timolia

uppercase folder + metadata

Bas950 reviewed (commented):
folder name didnt update...

l3nnartt committed:
Update and rename websites/T/timolia/Ttimoliapresence.ts to websites/T/Timolia/presence.ts

l3nnartt committed:
rename folder

l3nnartt committed:
Merge branch 'PreMiD:master' into master

l3nnartt committed:
fix

i am too stupid for git

l3nnartt commented:
i guess its fixed

Bas950 reviewed (changes requested):
Metadata in wrong one

Bas950 committed:
:truck:

Bas950 committed:
:truck:

Bas950 committed:
:art: Prettify
